java/server/test/org/openqa/selenium/grid/router/JmxTest.java

```
@@ -0,0 +1,222 @@
+
+      String usedSlots = (String) beanServer.getAttribute(name, "UsedSlots");
+      assertThat(Integer.parseInt(usedSlots)).isZero();
+
+      String load = (String) beanServer.getAttribute(name, "Load");
+      assertThat(Float.parseFloat(load)).isEqualTo(0.0f);
+
+      String remoteNodeUri = (String) beanServer.getAttribute(name, "RemoteNodeUri");
+      assertThat(remoteNodeUri).isEqualTo(nodeUri.toString());
+
+      String gridUri = (String) beanServer.getAttribute(name, "GridUri");
+      assertThat(gridUri).isEqualTo(nodeUri.toString());
+
+    } catch (InstanceNotFoundException | IntrospectionException | ReflectionException
+      | MalformedObjectNameException e) {
+      fail("Could not find the registered MBean");
+    } catch (MBeanException e) {
+      fail("MBeanServer exception");
+    } catch (AttributeNotFoundException e) {
+      fail("Could not find the registered MBean's attribute");
+    }
+  }
+
+  @Test
+  public void shouldBeAbleToRegisterSessionQueuerServerConfig() {
+    try {
+      ObjectName name = new ObjectName(
+        "org.seleniumhq.grid:type=Config,name=NewSessionQueueConfig");
+
+      new JMXHelper().unregister(name);
+
+      NewSessionQueueOptions queueOptions =
+        new NewSessionQueueOptions(new MapConfig(ImmutableMap.of()));
+      MBeanInfo info = beanServer.getMBeanInfo(name);
+      assertThat(info).isNotNull();
+
+      MBeanAttributeInfo[] attributeInfoArray = info.getAttributes();
+      assertThat(attributeInfoArray).hasSize(2);
+
+      String requestTimeout = (String) beanServer.getAttribute(name, "RequestTimeoutSeconds");
+      assertThat(Long.parseLong(requestTimeout)).isEqualTo(queueOptions.getRequestTimeoutSeconds());
+
+      String retryInterval = (String) beanServer.getAttribute(name, "RetryIntervalSeconds");
+      assertThat(Long.parseLong(retryInterval)).isEqualTo(queueOptions.getRetryIntervalSeconds());
+    } catch (InstanceNotFoundException | IntrospectionException | ReflectionException
```

It's fine to just let the test fail because the exception was thrown. That'll give you more information that just calling `fail` in these `catch` blocks.